pkg/testutil/prometheus.go

```diff
@@ -163,16 +163,16 @@ func (p *Prometheus) Start() error {
 		)
 	}
 	p.addr = fmt.Sprintf("localhost:%d", port)
-	p.cmd = exec.Command(
-		prometheusBin(p.version),
-		append([]string{
-			"--storage.tsdb.path=" + p.db.Dir(),
-			"--web.listen-address=" + p.addr,
-			"--web.route-prefix=" + p.prefix,
-			"--web.enable-admin-api",
-			"--config.file=" + filepath.Join(p.db.Dir(), "prometheus.yml"),
-		}, extra...)...,
-	)
+	args := append([]string{
+		"--storage.tsdb.retention=2d", // Pass retention cause prometheus since 2.8.0 don't show default value for that flags in web/api: https://github.com/prometheus/prometheus/pull/5433
```

Comment:
Just FYI: We need better handling for compatibility testing. Ideas are [here](https://github.com/improbable-eng/thanos/issues/758)

Feedback and help wanted (: